src/x_connective.c

```diff
@@ -1015,63 +1026,104 @@ static void trigger_list(t_trigger *x, t_symbol *s, int argc, t_atom *argv)
 {
     t_triggerout *u;
     int i;
-    for (i = (int)x->x_n, u = x->x_vec + i; u--, i--;)
+    for(i = x->x_n, u = x->x_vec + i; u--, i--;)
     {
         if (u->u_type == TR_FLOAT)
             outlet_float(u->u_outlet, (argc ? atom_getfloat(argv) : 0));
         else if (u->u_type == TR_BANG)
             outlet_bang(u->u_outlet);
         else if (u->u_type == TR_SYMBOL)
             outlet_symbol(u->u_outlet,
-                (argc ? atom_getsymbol(argv) : &s_symbol));
+                          (argc ? atom_getsymbol(argv) : (s != NULL ? s : &s_symbol)));
```

Comment:
are you sure about this? I think the original code makes sense (empty list -> "empty" symbol). I don't know if we want to output the "list" selector instead...

Reply:
ok, reverting back to original

Reply:
that came from purr data world, not my idea to be sure about it :) thanks for revising this